src/System.Text.Json/tests/WritableJsonApiTests.JsonNumberUnitTests.cs

```diff
@@ -0,0 +1,473 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Json
+{
+    public static partial class WritableJsonApiTests
+    {
+        [Theory]
+        [InlineData(-456)]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(long.MaxValue)]
+        [InlineData(2.3)]
+        [InlineData(-17.009)]
+        [InlineData(3.14f)]
+        [InlineData(0x2A)]
+        [InlineData(0b_0110_1010)]
+        [InlineData("1e400")]
+        [InlineData("1e+100000002")]
+        [InlineData("184467440737095516150.184467440737095516150")]
+        public static void TestJsonNumberConstructor(object value)
+        {
+            // should not throw any exceptions:
+            _ = new JsonNumber(value as dynamic);
+        }
+
+        [Fact]
+        public static void TestDefaultCtor()
+        {
+            var jsonNumber = new JsonNumber();
+            Assert.Equal(0, jsonNumber.GetInt32());
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(255)]
+        public static void TestByte(byte value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetByte(value);
```

Comment:
Also test with the custom ctor (not just SetByte).

There are 4 ways to create a JsonNumber:
1) Default ctor + Set
2) Numeric type ctor
3) Implicit cast
4) String ctor

We should exercise all 4 for each of the `Test{NumericType}` test here. We can then probably get rid of `TestImplicitOperators` which is only testing a particular value.

Same feedback elsewhere.

Reply:
So probably I should remove `TestJsonNumberConstructor` as well?

Reply:
Up to you. I would be fine with either. As long as some of the interesting cases from that are covered elsewhere. I did like how you used object/dynamic for the various test cases there.